### PR TITLE
MB-1536, MB-1319

### PR DESCRIPTION
**Pull request by ThilankaBowala**
MB-1536 : Cannot add sub topics from hierarchical view
MB-1319 : When "&" is in the queue or topic name, inconsistent behaviors were observed

Please merge with : https://github.com/wso2/andes/pull/453

### CODE REVIEW

File: components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java [hunk 1]
```diff
@@ -156,12 +155,6 @@ private void buildTopicTree(TopicNode topicNode, Collection resource, UserRegist
      */
     @Override
     public void addTopic(String topicName) throws EventBrokerException {
-        //Prevent creating topics, if topic name is not valid
-        if (!NameValidationUtils.isValidUITopicName(topicName)) {
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java]: Let's reformat this to if (!(NameValidationUtils.isValidUITopicName(topicName)))

File: components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java [hunk 2]
```diff
@@ -156,12 +155,6 @@ private void buildTopicTree(TopicNode topicNode, Collection resource, UserRegist
...
-            throw new EventBrokerException("Topic name " + topicName + " is not a valid topic name. Only alphanumeric"
-                    + " characters, stars(*), hash(#) and dots(.) are allowed. Dot can only use in the middle of the"
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java]: Shall we change the message to (.) can only be used in the middle...... and (#) can only be used at the end of the name.

File: components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/queues/js/treecontrol.js [hunk 3]
```diff
@@ -2,22 +2,22 @@
  * Adding queues, when creating queues from the user interface
  */
 function addQueue(createdFrom) {
-    var topic = document.getElementById("queue");
+    var queue = document.getElementById("queue");
 
     var error = "";
 
-    if (topic.value == "") {
+    if (queue.value == "") {
```
Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.ui/src/main/resources/web/queues/js/treecontrol.js]: let's flip this to if ("" == queue.value)

File: components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/js/treecontrol.js [hunk 4]
```diff
@@ -190,6 +190,10 @@ function addTopic() {
     topic.value = topic.value.replace(/^\s+|\s+$/g, "");
     if (topic.value == "") {
         error = "Topic name cannot be empty.\n";
+    }else if (!isValidTopicName(topic.value)) {
```
Comment (a5anka) [inline components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/js/treecontrol.js]: Need to have a space before else.
i.e.
 `} else`

File: components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/js/treecontrol.js [hunk 5]
```diff
@@ -214,7 +218,12 @@ function addTopicFromManage() {
 
     if (topic.value == "") {
         error = "Topic name cannot be empty.\n";
+    }else if (!isValidTopicName(topic.value)) {
```
Comment (a5anka) [inline components/andes/org.wso2.carbon.andes.event.ui/src/main/resources/web/topics/js/treecontrol.js]: Need to have a space before else.
i.e.
 `} else`

File: components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java (hunk 1 above)
Comment (a5anka) [inline components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java]: I'm afraid, we cannot format deleted code. :smile:

Comment (sdkottegoda) [inline components/andes/org.wso2.carbon.andes.event.core/src/main/java/org/wso2/carbon/andes/event/core/internal/subscription/registry/TopicManagerServiceImpl.java]: Oops sorry :D

**Pull request closed without merging**